### PR TITLE
DFS for Fence Planning

### PR DESCRIPTION
**Pull request by AAkashyap1**

### CODE REVIEW

File: solutions/usaco-944.mdx [hunk 1]
```diff
@@ -98,5 +98,88 @@ public class FencePlan {
...
+#define f first
+#define s second
+using namespace std;
+
+ifstream fin("fenceplan.in");
+ofstream fout("fenceplan.out");
+
+int n, m;
+
+vector<int> adj[100001];
```
Comment (SansPapyrus683) [inline solutions/usaco-944.mdx:125]: put 1e5 in a constant called `MAX_N`
and then when declaring this array use `MAX_N + 1`
or you could be a normal person and use nested vectors

File: solutions/usaco-944.mdx [hunk 2]
```diff
@@ -98,5 +98,88 @@ public class FencePlan {
...
+vector<bool> visited(100001, false);
```
Comment (SansPapyrus683) [inline solutions/usaco-944.mdx:126]: same here

File: solutions/usaco-944.mdx [hunk 3]
```diff
@@ -98,5 +98,88 @@ public class FencePlan {
...
+#include <algorithm>
+#include <vector>
+#define f first
+#define s second
+using namespace std;
+
+ifstream fin("fenceplan.in");
+ofstream fout("fenceplan.out");
+
+int n, m;
```
Comment (SansPapyrus683) [inline solutions/usaco-944.mdx:123]: these can be made local to `main`

File: solutions/usaco-944.mdx [hunk 4]
```diff
@@ -98,5 +98,88 @@ public class FencePlan {
...
+	for (int i = 0; i < m; i++) {
+		fin >> connections[i].f >> connections[i].s;
+	}
+	for (int i = 0; i < m; i++) {
+		adj[connections[i].f].push_back(connections[i].s);
+		adj[connections[i].s].push_back(connections[i].f);
+	}
+
+	connected_components();
+	long long min_perimeter = 10000000000;
```
Comment (SansPapyrus683) [inline solutions/usaco-944.mdx:164]: if all you need is a big number
use `INT64_MAX`

File: solutions/usaco-944.mdx [hunk 5]
```diff
@@ -98,5 +98,88 @@ public class FencePlan {
...
+}
+
+void connected_components() {
+	for (int i = 1; i < n + 1; i++) {
+		curr.resize(0);
+		if (visited[i]) continue;
+		dfs(i);
+		components.push_back(curr);
+	}
+}
```
Comment (SansPapyrus683) [inline solutions/usaco-944.mdx:146]: comments explaining what these 2 functions do?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
i don't know why you brought this upon yourself

**Review by nwatx — APPROVED** [review]

**Review by jessechoe10 — APPROVED** [review]

**Pull request closed without merging**